src/tools/rbd/action/MirrorImage.cc

```diff
@@ -187,6 +189,85 @@ int execute_resync(const po::variables_map &vm) {
   return 0;
 }
 
+void get_status_arguments(po::options_description *positional,
+			  po::options_description *options) {
+  at::add_image_spec_options(positional, options, at::ARGUMENT_MODIFIER_NONE);
+  at::add_format_options(options);
+}
+
+int execute_status(const po::variables_map &vm) {
+  at::Format::Formatter formatter;
+  int r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string image_name;
+  std::string snap_name;
+  r = utils::get_pool_image_snapshot_names(
+      vm, at::ARGUMENT_MODIFIER_NONE, &arg_index, &pool_name, &image_name,
+      &snap_name, utils::SNAPSHOT_PRESENCE_NONE);
+  if (r < 0) {
+    return r;
+  }
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  librbd::Image image;
+  r = utils::init_and_open_image(pool_name, image_name, "", false,
+                                 &rados, &io_ctx, &image);
+  if (r < 0) {
+    return r;
+  }
+
+  std::string image_id = utils::image_id(image);
+
+  cls::rbd::MirrorImage mirror_image;
+  r = librbd::cls_client::mirror_image_get(&io_ctx, image_id, &mirror_image);
+  if (r < 0) {
+    std::cerr << "rbd: failed to get global image id for image " << image_name
+	      << ": " << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  if (mirror_image.global_image_id.empty()) {
+    std::cerr << "rbd: failed to get global image id for image " << image_name
+	      << std::endl;
+    return -EINVAL;
+  }
+
+  cls::rbd::MirrorImageStatus status;
+  r = librbd::cls_client::mirror_image_status_get(&io_ctx,
+						  mirror_image.global_image_id,
+						  &status);
+  if (r < 0) {
+    std::cerr << "rbd: failed to get status for image " << image_name << ": "
+	      << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  if (formatter != nullptr) {
+    formatter->open_object_section("image");
+    formatter->dump_string("image_name", image_name);
```

Comment:
Minor: probably safe to drop the "image_" prefix here and below since you are within an "image" object section -- implied.